code/game/objects/structures/displaycase.dm

```diff
@@ -106,13 +106,25 @@
 	return
 
 /obj/structure/displaycase/attackby(obj/item/weapon/W, mob/user, params)
-	if(W.GetID() && electronics && !broken)
+	if(W.GetID() && !broken)

```

Comment:
that doesn't fix it, you're just making it openable by everyone if it has electronics.
To be honest I'm not even sure if there's even a displaycase that shouldn't be openable, it's been a while since I last read displaycase attackby code.